### PR TITLE
Add link checking

### PR DESCRIPTION
**Pull request by igorkorenfeld**
## Changes proposed in this pull request:

This PR adds automated link checking to our GitHub workflow, using the `html-check-links` [package](https://rocket.modern-web.dev/tools/check-html-links/overview/#usage-examples).  This test checks only internal links. 

`html-check-links` runs quickly (for internal links), and generates helpful output. However due and existing bug the current version returns a success code even when there are a missing/broken links. To get around that issue, the `npm` script has a `grep` search for a checkmark to test for success. Definitely open to suggestions on how to improve that.

### CODE REVIEW

**Comment by igorkorenfeld** [comment]
It seems like external url checks are not working as expected. Example:
```
 81. missing reference target _site/product/define/vision/index.html
      from _site/product/define/vision/index.html:49:14 via href="https://unpkg.com/prismjs@1.20.0/themes/prism.css"
      from _site/product/define/vision/index.html:430:12 via href="https://www.prodpad.com/blog/product-vision-template/"
      from _site/product/define/vision/index.html:433:12 via href="https://app.mural.co/t/gsa6/m/gsa6/1614283990848/1340f39d85056094e7e7c96897587f835b4c5138"
      ... 4 more references to this target
```

The `href`s in this example do not appear to point to `_site/product/define/vision/index.html` nor are they broken themselves (i.e. they resolve with a 200). I am not sure why the page itself gets flagged as a missing target.

**Comment by igorkorenfeld** [comment]
@beechnut this should be ready for review now.

**Comment by beechnut** [comment]
@igorkorenfeld I see a lot of broken internal links, mostly in the Engineering Guide. I'd prefer we get the build passing before merging so we don't break everyone else's builds.

I've made progress on a number of the links — let me know if I should add the commits to this branch or if you'd prefer a different strategy.

**Comment by igorkorenfeld** [comment]
Agree with that! Thanks for making progress on those links. Maybe let's combine those with the fixes I made on here: https://github.com/18F/approaches/tree/ik/fix-broken-links-1 . Then we can merge that in to this branch?

**Comment by beechnut** [comment]
@igorkorenfeld FYI I'm making quick progress on this, just a few links to go and I'll push my work.

**Review by beechnut — APPROVED** [review]

File: docs/replatforming.md [hunk 1]
```diff
@@ -115,3 +121,12 @@ _Example:_
 ```
 <span style = "color:#58AA02" class="exampleFailure" data-pa11y-ignore>This text fails. </span>
 ```
+
+### HTML Validation
+`html-validate` will check for valid HTML. It is configured in `.htmlvalidate.json`.
+
+### Link checking
+`check-html-links` will test both internal and external links on the site. The internal link check tests whether a target link file exists in the `_site` folder at the expected location. Because the current version of `check-html-link` [does not return an error value](https://github.com/modernweb-dev/rocket/issues/166) when it finds broken links, the npm script for this check includes an additional grep search for a "✅" which would appear only if there are no broken links. With this (hopefully) temporary fix in place, github actions will report a failure if there are broken links.
```
[the remark this answers is not in the document]
Comment (igorkorenfeld) [reply docs/replatforming.md]: Good catch, I thought I removed all of those references but I didn't. Updated to remove them now and add explanation on alternative `npm` command to run the check locally.

**Pull request merged**